components/common/hooks/useGetSubscription.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {
+    getCloudSubscription as getCloudSubscriptionAction,
+} from 'mattermost-redux/actions/cloud';
+
+import {Subscription} from '@mattermost/types/cloud';
+import {getCloudSubscription} from 'mattermost-redux/selectors/entities/cloud';
+
+export default function useGetSubscription(): Subscription | undefined {
+    const cloudSubscription = useSelector(getCloudSubscription);
+    const retrievedCloudSub = Boolean(cloudSubscription);
+    const dispatch = useDispatch();
+    const [requestedSubscription, setRequestedSubscription] = useState(false);
+
+    useEffect(() => {
+        if (!retrievedCloudSub && !requestedSubscription) {
```

Comment:
Should we check here that they have the cloud license? We may get the 500s we were seeing in community here too

Reply:
Added a check for it

Reply:
I think the other limits/usage hooks are going to need this addition - right now they hinge on the feature flag being set, but eventually that will go away